### PR TITLE
fix(scss): updated to current scss spec, updated deps

### LINKED ISSUE #6916: SASS deprecation warnings
When using the sass files via sass-embedded (from sass-loader in webpack) I see deprecation warnings like this:
```
WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 69, column 2 of file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss:69:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

69 |   margin-inline-end: var(--#{$breadcrumb}__item-divider--MarginInlineEnd);


file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss 70:3  @use
file:///home/dev/oversight/src/oversight/static_src/oversight/vendor.scss 22:1                             root stylesheet

 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 70, column 2 of file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss:70:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

70 |   font-size: var(--#{$breadcrumb}__item-divider--FontSize);


file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss 71:3  @use
file:///home/dev/oversight/src/oversight/static_src/oversight/vendor.scss 22:1                             root stylesheet

 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 71, column 2 of file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss:71:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

71 |   line-height: 1;


file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss 72:3  @use
file:///home/dev/oversight/src/oversight/static_src/oversight/vendor.scss 22:1                             root stylesheet

 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 72, column 2 of file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss:72:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

72 |   color: var(--#{$breadcrumb}__item-divider--Color);


file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Breadcrumb/breadcrumb.scss 73:3  @use
file:///home/dev/oversight/src/oversight/static_src/oversight/vendor.scss 22:1                             root stylesheet

 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 308, column 8 of file:///home/dev/oversight/node_modules/@patternfly/patternfly/sass-utilities/mixins.scss:308:8:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

308 |         #{$prop}: #{$variable-list};


file:///home/dev/oversight/node_modules/@patternfly/patternfly/sass-utilities/mixins.scss 309:9  @content
file:///home/dev/oversight/node_modules/@patternfly/patternfly/sass-utilities/mixins.scss 235:5  pf-v6-apply-breakpoint()
file:///home/dev/oversight/node_modules/@patternfly/patternfly/sass-utilities/mixins.scss 303:5  pf-v6-build-css-variable-stack()
file:///home/dev/oversight/node_modules/@patternfly/patternfly/components/Brand/brand.scss 15:3  @use
file:///home/dev/oversight/src/oversight/static_src/oversight/vendor.scss 21:1                   root stylesheet

 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

WARNING in ./src/oversight/static_src/oversight/vendor.scss (./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./src/oversight/static_src/oversight/vendor.scss)
Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
210 repetitive deprecation warnings omitted.
Run in verbose mode to see all warnings.
 @ ./src/oversight/static_src/oversight/vendor.scss 8:6-159 20:17-24 24:7-21 50:25-39 51:36-47 51:50-64 55:6-65:7 56:54-65 56:68-82 62:42-53 62:56-70 64:21-28 75:0-129 75:0-129 76:22-29 76:33-47 76:50-64 53:4-66:5
 @ ./src/oversight/static_src/oversight/vendor.js 1:0-23

webpack 5.93.0 compiled with 6 warnings in 1448 ms
```

### PR DESCRIPTION
**Pull request by mattnolting**
closes #6916

# TLDR
Our sass configuration and dependencies are deprecated. There are several issues throwing build errors. 
1) `@include` statements at the top of CSS declaration blocks
2) Outdated `renderSync()` function compilation function
3) Out of date or missing transitive dependencies
4) Stylelint order errors
5) SASS library calls (`map-get()`-> `map.get()`, `type-of()` -> `list.type-of()`, etc)

**This PR:**
1) Fixes CSS inclusion order (`@media`, `@include`, 
2) Updates `renderSync()` to `sass.compile()`
3) Update transitive dependencies
4) Resolves stylelint errors
5) Calls appropriate SASS libs

Looks like we need to update backstop refs. Here's the current regression report. [Backstop report](https://drive.google.com/file/d/1vVn1BdRJeiG6sThWNFXL42A-GS9PCUNj/view?usp=sharing)

### CODE REVIEW

**Review by mcoker — COMMENTED** [review]
Not quite done with the review, but wanted to leave a comment:
* I had a question below
* Looks like I don't have permission to view the [backstop report](https://drive.google.com/file/d/1vVn1BdRJeiG6sThWNFXL42A-GS9PCUNj/view?usp=sharing)
* created a diff report of patternfly.css from this PR and on main (https://www.diffchecker.com/4ZIWu5tj/)
* looks like the build is failing possibly from this error _"error rimraf@6.0.1: The engine "node" is incompatible with this module. Expected version "20 || >=22". Got "18.20.4"
"_ @dlabaj wdyt about the build failure?

**Comment by mcoker** [comment]
Cleaning up the PR list and closing this out of date one. We've since updated sass to latest and made the style changes needed and get no warnings or anything 🎉
